presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/jmx/StatisticsAwareConnectionFactory.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.jmx;
+
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcIdentity;
+import org.weakref.jmx.Flatten;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static java.util.Objects.requireNonNull;
+
+public class StatisticsAwareConnectionFactory
+        implements ConnectionFactory
+{
+    private final ConnectionFactoryStats stats = new ConnectionFactoryStats();
```

Comment:
We could simplify this by inlining the stats fields, since there are only two of them. Up to you.